RHDEVDOCS-4473: rel notes odo 3.0 (#51902)
openshift/openshift-docs: cli_reference/developer_cli_odo/odo-release-notes.adoc

:_content-type: ASSEMBLY
[id='odo-release-notes']
= `{odo-title}` 3.0.0 release notes
include::_attributes/common-attributes.adoc[]
:context: odo-release-notes

toc::[]

[id="odo-notable-improvements_{context}"]
== Notable changes and improvements in `{odo-title}` version 3.0.0

> `odo` 3.0 includes new commands to improve the developer experience, providing the same functionality as before, but in a faster and more efficient manner.

Reviewer: ```suggestion
`odo` 3.0 includes new commands to improve the developer experience. These commands provide the same functionality as before but in a faster and more efficient manner.
```
- Rewrite to clarify the unclear modifier of "providing". As constructed, the sentence isn't clear what is providing the same functionality -- the experience or the commands.
- no comma needed before a conjunction with a dependent clause.

For more information, see the section "Major highlights of odo 3.0" in xref:./understanding-odo.adoc#understanding-odo[Understanding odo].

[id="odo-fixed-issues_{context}"]

> == Bug fixes in `{odo-title}` version 3.0.0

Reviewer: Items in the bug list should use parallel constructions. Each item should be a complete sentence or should be an incomplete sentence, but the two shouldn't mixed.

> * `odo` does not honor directory `.gitignore` entries like `/target`.

Reviewer: ```suggestion
* `odo` does not honor directory `.gitignore` entries like `/target`.
```
Is this the bug or the fix? Consider providing more detail in 2 sentences using the "Before this update" > "With this release" construction

> * Port names should match endpoint names.

Reviewer: ```suggestion
* Port names must match endpoint names.
```
Avoid `should`. It sounds too equivocating. Use `must` if it's a requirement

* `odo deploy` - devfile variable substitution does not occur in URI-referenced deployment manifest YAML file.

* `odo init --name <name> --devfile-path <path>` panics when a non-existent path is used.

* Commands should get namespace from `env` file.

> * `odo dev` fails after component was deployed using `odo deploy`.

Reviewer: ```suggestion
* `odo dev` fails after a component was deployed using `odo deploy`.
```

> * `odo dev` does not use `.gitignore` to sync files to container.

Reviewer: ```suggestion
* `odo dev` does not use `.gitignore` to sync files to a container.
```

> * `odo deploy` and `odo build-images` not working on Windows.

Reviewer: ```suggestion
* `odo deploy` and `odo build-images` do not work on Windows operating systems.
```
Per ISG, refer to `Windows operating system` or `Windows system`

* `odo delete` fails for Kubernetes components.

> * Add support for `command` field in container components.

Reviewer: Rewrite as a complete sentence

> * Stopping `odo dev` also removes `odo deploy` resources.

Reviewer: Does this mean killing the process or something else?

> * Configuration shows volumes as containers

Reviewer: This item needs more detail and needs to use the template we have for known issue items

> * Do not save component name in `env.yaml`.

Reviewer: Is this a known issue? Rewrite to use the format provided in the known issues template.

> * Do not save component name in `env.yaml`.

Reviewer: ```suggestion
* Component names are not saved in `env.yaml`.
```

This phrasing sound like something we are asking users to do. Rewrite to use the format provided in the bug fixes template.

> * Failed to delete the `ServiceBinding` resource.

Reviewer: Same comment for this and for all the rest of the items in the list: rewrite using the bug fix template or similar phrasing.

For more information, see the section "Major highlights of odo 3.0" in xref:./understanding-odo.adoc#understanding-odo[Understanding odo].

[id="odo-fixed-issues_{context}"]
== Bug fixes in odo version 3.0

* Before this update, `odo` did not honor directory entries in the `.gitignore` file, for example, `/target`. With this update, `odo` honors all `.gitignore` entries. 
link:https://github.com/redhat-developer/odo/issues/4389[(odo-4389)]

* Before this update, port names did not match endpoint names. With this update, `odo` ensures that the names match.
link:https://github.com/redhat-developer/odo/issues/4737[(odo-4737)]

* Before this update, when running the `odo deploy` command, variables defined in a devfile would not be substituted in URI-referenced deployment manifest YAML files. With this update, `odo` performs the variable substitution correctly.
link:https://github.com/redhat-developer/odo/issues/5451[(odo-5451)]

* Before this update, running the command `odo init --name <name> --devfile-path <path>` caused a panic to occur with a non-existent path. This update fixes the issue.
link:https://github.com/redhat-developer/odo/issues/5540[(odo-5540)]

> * Before this update, `odo` commands ignored namespace settings in the `.odo/env/env.yaml` file. With this update, `odo` applies the namespace setting.

Reviewer: ```suggestion
* Before this update, `odo` commands ignored namespace settings in the `.odo/env/env.yaml` file. With this update, running the `odo` command applies the namespace setting.
```

Author: there is no singular `odo` command - this is referring to the `odo` product that applies the namespace for all odo commands so I think it is ok as is

Reviewer: ack

link:https://github.com/redhat-developer/odo/issues/5544[(odo-5544)]

> * Before this update, running the `odo dev` command could fail after you deployed a component with `odo deploy`. This update fixes the issue.

Reviewer: ```suggestion
* Before this update, running the `odo dev` command could fail after you deployed a component with the `odo deploy` command. This update fixes the issue.
```

link:https://github.com/redhat-developer/odo/issues/5546[(odo-5546)]

> * Before this update, the `odo dev` command did not use the `.gitignore` file when synchronizing files to a container. With this update, `odo` uses the `.gitignore` file.

Reviewer: ```suggestion
* Before this update, the `odo dev` command did not use the `.gitignore` file when synchronizing files to a container. With this update, the `odo dev` command uses the `.gitignore` file.
```

Author: I think it reads better saying `odo` fixes it , rather than having repetition with "the `odo dev` command", so going to leave this as is.

link:https://github.com/redhat-developer/odo/issues/5588[(odo-5588)]

* Before this update, the `odo delete` command failed for Kubernetes components. This update fixes the issue.
link:https://github.com/redhat-developer/odo/issues/5637[(odo-5637)]

* With this update, `odo` now supports the `command` field in container components.
link:https://github.com/redhat-developer/odo/issues/5648[(odo-5648)]

* Before this update, when you stopped the `odo dev` process, resources created using `odo deploy` were removed. With this update, `odo` only deletes resources that were deployed using `odo dev`.
link:https://github.com/redhat-developer/odo/issues/5700[(odo-5700)]

* Before this update, `odo` would display volumes defined in the devfile as containers. This update fixes the issue.
link:https://github.com/redhat-developer/odo/issues/5779[(odo-5779)]

* Before this update, `odo` would save the component name in the `env.yaml` file. With this update, the component name is only specified in the devfile.
link:https://github.com/redhat-developer/odo/issues/5780[(odo-5780)]

> * Before this update, after terminating the `odo dev` command, `odo` would delete the `ServiceBinding` resource but the logs would have an entry stating "Failed to delete the `ServiceBinding` resource". This update fixes the issue.

Reviewer: ```suggestion
* Before this update, after terminating the `odo dev` command, `odo` would delete the `ServiceBinding` resource, but the logs would have an entry stating `Failed to delete the ServiceBinding resource`. This update fixes the issue.
```
Per the ISG, elements like log entries should be in monospace (i.e. backticks) and not quotation marks

link:https://github.com/redhat-developer/odo/issues/5797[(odo-5797)]

* Before this update, if you ran the `odo dev` command on a white background terminal, the colours could look wrong or the content could be hard to read. With this update, the content is displayed correctly.
link:https://github.com/redhat-developer/odo/issues/5843[(odo-5843)]

* Before this update, the `odo logs` command could be slow to respond in certain environments. With this update, the command responds as expected.
link:https://github.com/redhat-developer/odo/issues/5872[(odo-5872)]

> * Before this update, the `odo dev port forward` command could timeout if the application is not ready. This update fixes the issue.

Reviewer: ```suggestion
* Before this update, the `odo dev port forward` command might timeout if the application is not ready. This update fixes the issue.
```
Per ISG: Don't use "could" as a synonym for "might" or "can".

Author: I think "could" is ok for past tense, but "might" works as well so will change here

link:https://github.com/redhat-developer/odo/issues/5877[(odo-5877)]

> * Before this update, the `odo dev` command would output the message `✗ Executing the application`, after cleaning up the resources. With this update, the message is no longer output.

Reviewer: ```suggestion
* Before this update, the `odo dev` command would output the message `Executing the application`, after cleaning up the resources. With this update, the message is no longer output.
```
For localization and screen reader purposes, avoid using icons

link:https://github.com/redhat-developer/odo/issues/5881[(odo-5881)]

* Before this update, `odo` would not deploy `ServiceBinding` resources when you ran the `odo deploy` command. With this update, `ServiceBinding` resources are deployed correctly.
link:https://github.com/redhat-developer/odo/issues/5883[(odo-5883)]

> 

Reviewer: ```suggestion
```

* Before this update, if the `Ephemeral` preference was set to `false`, the persistent volume claim (PVC) would not owned by the `Deployment` resource. This update fixes the issue.
link:https://github.com/redhat-developer/odo/issues/5886[(odo-5886)]

> 

Reviewer: ```suggestion
```

> * Before this update, `odo` would send telemetry data even if the `ODO_DEBUG_TELEMETRY_FILE` environment variable was set. With this update, `odo` saves all telemetry data to file when the variable is set.

Reviewer: ```suggestion
* Before this update, `odo` would send telemetry data even if the `ODO_DEBUG_TELEMETRY_FILE` environment variable was set. With this update, `odo` saves all telemetry data to a file when the variable is set.
```
